src/test/java/org/prebid/server/ApplicationTest.java

```diff
@@ -246,193 +489,543 @@ public void openrtb2AuctionShouldRespondWithBidsFromDifferentExchanges() throws
+
+        final Map<String, String> exchanges = new HashMap<>();
+        exchanges.put(SOMOAUDIENCE, "http://localhost:" + WIREMOCK_PORT + "/somoaudience-exchange");
+
+        //when
+        final Response response = given(spec)
+                .header("Referer", "http://www.example.com")
+                .header("X-Forwarded-For", "192.168.244.1")
+                .header("User-Agent", "userAgent")
+                .header("Origin", "http://www.example.com")
+                .cookie("uids", "eyJ1aWRzIjp7InNvbW9hdWRpZW5jZSI6IlNNLVVJRCJ9fQ==")
+                .body(jsonFrom("openrtb2/somoaudience/test-auction-somoaudience-request.json"))
+                .post("/openrtb2/auction");
+
+        //then
+        final String expectedAuctionResponse = auctionResponseFrom(jsonFrom("openrtb2/somoaudience/test-auction-somoaudience-response.json"),
+                response,  "ext.responsetimemillis.%s", exchanges);
+
+        JSONAssert.assertEquals(expectedAuctionResponse, response.asString(), JSONCompareMode.NON_EXTENSIBLE);
+    }
+
+    @Test
+    public void openrtb2AuctionShouldRespondWithBidsFromBeachfront() throws IOException, JSONException {
+        // given
         // beachfront bid response for imp 18
         wireMockRule.stubFor(post(urlPathEqualTo("/beachfront-video-exchange"))
                 .withQueryParam("exchange_id", equalTo("beachfrontAppId"))
                 .withQueryParam("prebidserver", equalTo(""))
                 .withHeader("Content-Type", equalToIgnoreCase("application/json;charset=UTF-8"))
                 .withHeader("Accept", equalTo("application/json"))
-                .withRequestBody(equalToJson(jsonFrom("openrtb2/test-beachfront-bid-request-1.json")))
-                .willReturn(aResponse().withBody(jsonFrom("openrtb2/test-beachfront-bid-response-1.json"))));
+                .withRequestBody(equalToJson(jsonFrom("openrtb2/beachfront/test-beachfront-bid-request-1.json")))
+                .willReturn(aResponse().withBody(jsonFrom("openrtb2/beachfront/test-beachfront-bid-response-1.json"))));
+
+        // pre-bid cache
+        wireMockRule.stubFor(post(urlPathEqualTo("/cache"))
+                .withRequestBody(equalToJson(jsonFrom("openrtb2/beachfront/test-cache-beachfront-request.json")))
+                .willReturn(aResponse().withBody(jsonFrom("openrtb2/beachfront/test-cache-beachfront-response.json"))));
+
+        final Map<String, String> exchanges = new HashMap<>();
+        exchanges.put(BEACHFRONT, "http://localhost:" + WIREMOCK_PORT + "/beachfront-video-exchange?exchange_id=");
+
+        //when
+        final Response response = given(spec)
+                .header("Referer", "http://www.example.com")
+                .header("X-Forwarded-For", "192.168.244.1")
+                .header("User-Agent", "userAgent")
+                .header("Origin", "http://www.example.com")
```

Comment:
There's no uids cookie - is this intentional?

Reply:
no

Reply:
Please address this comment